pydata_sphinx_theme/__init__.py

```diff
@@ -167,6 +178,66 @@ def get_edit_url():
 # -----------------------------------------------------------------------------
 
 
+def setup_cdn(app, pagename, templatename, context, doctree):
+    """hoist the `use_public_cdns` config value to the template context
+    """
+
+    use_public_cdns = app.config.use_public_cdns
+
+    context["use_public_cdns"] = use_public_cdns
+
+    if not has_default_mathjax_path(app):
+        return
+
+    if at_least_sphinx_three or use_public_cdns:
+        return
+
+    mathjax_path = app.config.mathjax_path
+
+    if not warned.get("mathjax_path"):
+        logger.warning(
+            "`use_public_cdns` is %s, but `mathjax_path` is configured,"
+            " probably by default, as:\n\n"
+            "  %s\n\n"
+            "> upgrade to `sphinx >=3`, which supports event `priority`...\n"
+            "> or configure `mathjax_path` in `conf.py`, e.g.:\n\n"
+            "  import %s\n"
+            "  def setup(app):\n"
+            "      app.config.use_public_cdns = False\n"
+            "      app.config.mathjax_path = %s.get_mathjax_path()\n",
+            use_public_cdns,
+            mathjax_path,
+            NAME,
+            NAME,
+            type="configuration",
+            location=pagename,
+        )
+        warned["mathjax_path"] = True
```

Comment:
@bollwyvl can you explain the mathjax situation a bit more?

So sphinx has an option to set the mathjax path (https://www.sphinx-doc.org/en/master/usage/extensions/math.html#confval-mathjax_path). But if we use our vendored assets (which included mathjax), we override what a user might have configured for mathjax?

Reply:
Again, it's all rather mysterious to me as well!

The math extension adds this [configurable](https://github.com/sphinx-doc/sphinx/blob/3.x/sphinx/ext/mathjax.py#L98-L100) and then at `env-updated`, it adds the mathjax link. I don't know how to intercept it doing that, hence this whole rube goldberg setup.

In `sphinx <3` (which i assume needs to be supported) the only way to control the order is the actual `extensions` order in `setup.py`, but themes are "special" and don't go there. So it checks whether `mathjax_path` is set to the default `cloudflare` value (which is luckily kept around), and will replace _only that_. 

As a user, one can set it to whatever is wanted (e.g. for SVG output) in `conf.py` and it would still be respected. But vendoring _all_ of MathJax is 75mb, so i just tried to bring in everything that was related to the default output, so if one _did_ want SVG, they'd have to bring all the assets into their `_static`. This would probably need documentation.